src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala

```diff
@@ -321,26 +322,21 @@ final case class ClientsApiServiceImpl(
   private def decorateKey(
     key: AuthorizationProcessModel.Key
   )(implicit contexts: Seq[(String, String)]): Future[PublicKey] = {
-    for {
-      user         <- userRegistryService.findById(key.relationshipId)
-      decoratedKey <- decorateKeyInternal(key, user)
-    } yield decoratedKey
-  }
-
-  private def decorateKeyInternal(key: AuthorizationProcessModel.Key, user: UserResource)(implicit
-    contexts: Seq[(String, String)]
-  ): Future[PublicKey] = {
-
-    val result: Future[PublicKey] = for {
+    val result = for {
       relationship     <- partyProcessService.getRelationship(key.relationshipId)
+      user             <- userRegistryService.findById(relationship.from)
       relationshipInfo <- PartyProcessConverter.toApiRelationshipInfo(user, relationship)
     } yield relationshipInfo.state match {
       case RelationshipState.ACTIVE => key.toApi(isOrphan = false, user)
-      case _                        => key.toApi(isOrphan = true, user)
+      case _                        =>
+        // It could be that in the new API, when a user is removed, a 404 error is returned.
+        // At this moment, it should return the relationship with the state set to Deleted.
+        key.toApi(isOrphan = true, user)
     }
 
     result.recoverWith {
-      case PartyProcessApiError(404, _, _, _, _) => Future.successful(key.toApi(isOrphan = true, user))
+      case PartyProcessApiError(404, _, _, _, _) =>
+        Future.successful(key.toApi(isOrphan = true, UserResource(id = UUID.randomUUID())))
```

Comment:
Would it be better to use the `relationshipId` here instead of a random UUID as suggested by @galales ? @beetlecrunch